src/lib.rs

```diff
@@ -271,20 +261,22 @@ impl Mnemonic {
 		for (idx, word) in words.enumerate() {
 			// Scrap languages that don't have this word.
 			for (i, lang) in langs.iter().enumerate() {
-				if possible[i] {
-					possible[i] = lang.find_word(word).is_some();
-				}
+				possible[i] &= lang.find_word(word).is_some();
 			}
 
-			// If there is just one language left, return it.
-			let nb_possible = possible.iter().filter(|p| **p).count();
-			if nb_possible == 1 {
-				return Ok(*possible.iter().zip(langs.iter()).find(|&(p, _)| *p).map(|(_, l)| l).unwrap());
-			}
-
-			// If all languages were eliminated, it's an invalid word.
-			if nb_possible == 0 {
-				return Err(Error::UnknownWord(idx));
+			// Get an iterator over remaining possible languages.
+			let mut iter = possible.iter().zip(langs.iter()).filter(|(p, _)| **p).map(|(_, l)| l);
+
+			match iter.next() {
+				// If all languages were eliminated, it's an invalid word.
+				None => return Err(Error::UnknownWord(idx)),
```

Comment:
Minor nit, please don't change it unless you deem it important yourself: technically it doesn't have to be the current word that was "invalid", it could have been any word before that, that happened to exclude some language that was actually meant. The common case is "this is an invalid word" though, so it makes a lot of sense to not overcomplicate it.

Reply:
Good point. I agree that this is the best way to return it though. Majority of cases an accidental invalid word will not by chance be valid in a random wrong language :sweat_smile: